deepchecks/tabular/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -114,14 +112,21 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         dataset.assert_features()
         dataset = dataset.sample(self.n_samples, random_state=self.random_state, drop_na_label=True)
         predictions = context.model.predict(dataset.features_columns)
-        y_proba = context.model.predict_proba(dataset.features_columns) if \
-            context.task_type in [TaskType.MULTICLASS, TaskType.BINARY] else None
+        if context.task_type in [TaskType.MULTICLASS, TaskType.BINARY]:
+            y_proba = context.model.predict_proba(dataset.features_columns)
+            # If proba shape does not match label, raise error
+            if y_proba.shape[1] != len(context.classes):
```

Comment:
this should not be here!
should be validated in context

Reply:
It's only here because here we use the proba directly and not through scorer, so we can't allow for new labels.
In the general case I don't want to force `y_proba.shape[1] == len(context.classes)` because it prevents new labels